Project doesn't compile with Java 20

**Describe the bug**
Kotlin doesn't yet support Java 20, so Gradle doesn't support it either yet.
See: https://github.com/gradle/gradle/issues/23488

**Steps to reproduce**
Using a Java 20 release, try building the project. `./gradlew compileJava` for example.

**What did you see?**
```
> Task :conventions:compileKotlin FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':conventions:compileKotlin'.
> Error while evaluating property 'compilerOptions.jvmTarget' of task ':conventions:compileKotlin'.
   > Failed to calculate the value of property 'jvmTarget'.
      > Unknown Kotlin JVM target: 20
```

**Additional context**
I don't expect us to fix anything right now.  Just documenting the problem as [CONTRIBUTING.md](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/CONTRIBUTING.md) suggests Java 17+ is required.  Perhaps we should specify an upper range of what has been tested?

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -8,7 +8,7 @@ and discuss your ideas or propose the changes you wish to make.
 
 ## Building
 
-In order to build and test this whole repository you need JDK 17 or higher.
+This project requires Java 17 to build and run tests. Newer JDK's may work, but this version is used in CI.
 
 Some instrumentations and tests may put constraints on which java versions they support.
 See [Running the tests](./docs/contributing/running-tests.md) for more details.
@@ -22,7 +22,7 @@ the Sonatype OSS snapshots repository at `https://oss.sonatype.org/content/repos
 
 ### Building from source
 
-Build using Java 17+:
+Build using Java 17:
 
 ```bash
 java -version

